### DISCUSSION TITLE
Can tests be rewired programmatically with the TestingEngine?

### DISCUSSION THREAD

**Discussion by jhwj9617** [microsoft/coyote#390] [Q&A] [answered]
I noticed that when running with the TestingEngine, I haven't found a way to rewire from there.
Instead, I have to create a post-build process that runs the `coyote rewrite` step, before the systematic unit tests runs.
Am I missing some function of TestingEngine, or is the way I'm doing it the right way?

**Comment by akashlal** [comment, marked as answer]
This is the right way. You still have to do `coyote rewrite` even if you are running the tests programmatically.

**Comment by jhwj9617** [comment]
Thanks for the clarity. Maybe convey this info in the docs?

**Comment by pdeligia** [comment]
> Thanks for the clarity. Maybe convey this info in the docs?

We will, thanks for the feedback @jhwj9617

**Comment by jhwj9617** [comment]
Just wanted to also point out [this ](https://github.com/microsoft/coyote/issues/179)would directly handle this which you are already addressing:
